### PR TITLE
backport: trivial 2024 10 23 pr8

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Batch of trivial backports

## What was done?
See commits

## How Has This Been Tested?
built locally; large combined merge passed tests locally

## Breaking Changes
Should be none

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: src/support/lockedpool.cpp [hunk 1]
```diff
@@ -27,7 +27,6 @@
 #include <iomanip>
 #include <iostream>
 #endif
-#include <utility>
```
Comment (UdjinM6) [inline src/support/lockedpool.cpp]: 27151: not really a duplicate in our case - consider backporting 26924 which adds the second one or drop this backport

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -2744,6 +2744,10 @@ static RPCHelpMan scantxoutset()
         "    combo(<pubkey>)                      P2PK and P2PKH outputs for the given pubkey\n"
         "    pkh(<pubkey>)                        P2PKH outputs for the given pubkey\n"
         "    sh(multi(<n>,<pubkey>,<pubkey>,...)) P2SH-multisig outputs for the given threshold and pubkeys\n"
+        "    tr(<pubkey>)                         P2TR\n"
+        "    tr(<pubkey>,{pk(<pubkey>)})          P2TR with single fallback pubkey in tapscript\n"
+        "    rawtr(<pubkey>)                      P2TR with the specified key as output key rather than inner\n"
+        "    wsh(and_v(v:pk(<pubkey>),after(2)))  P2WSH miniscript with mandatory pubkey and a timelock\n"
```
Comment (UdjinM6) [inline src/rpc/blockchain.cpp:2750]: 27155: DMN, we don't have any of these

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/support/lockedpool.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/support/lockedpool.cpp]: dropped

File: src/rpc/blockchain.cpp (hunk 2 above)
Comment (PastaPastaPasta) [reply src/rpc/blockchain.cpp]: dropped

**Review by UdjinM6 — APPROVED** [review]
utACK bc6ffc6bdbb213e7622b1c989b657869ce57aad8

should probably wait for guix to confirm

File: contrib/verifybinaries/README.md [hunk 3]
```diff
@@ -1,16 +1,5 @@
 ### Verify Binaries
 
-#### Preparation:
```
Comment (knst) [inline contrib/verifybinaries/README.md:3]: 27107: just a note: maybe we want to dashify `contrib/verifybinaries/README.md` someday?

File: src/util/system.h [hunk 4]
```diff
@@ -294,15 +294,13 @@ class ArgsManager
      * Get data directory path
      *
      * @return Absolute path on success, otherwise an empty path when a non-directory path would be returned
```
Comment (knst) [inline src/util/system.h:296]: 27176: we didn't backport 27073, which is dependency for this backport

Comment (PastaPastaPasta) [reply src/util/system.h]: dropped

File: doc/developer-notes.md [hunk 5]
```diff
@@ -842,12 +842,12 @@ Strings and formatting
...
     tend to act differently depending on platform, or even the user locale.
 
-- Use `ParseInt32`, `ParseInt64`, `ParseUInt32`, `ParseUInt64`, `ParseDouble` from `utilstrencodings.h` for number parsing.
+- Use `ToIntegral` from [`strencodings.h`](/src/util/strencodings.h) for number parsing. In legacy code you might also find `ParseInt*` family of functions, `ParseDouble` or `LocaleIndependentAtoi`.
 
   - *Rationale*: These functions do overflow checking and avoid pesky locale issues.
 
 - Avoid using locale dependent functions if possible. You can use the provided
-  [`lint-locale-dependence.sh`](/test/lint/lint-locale-dependence.sh)
+  [`lint-locale-dependence.py`](/test/lint/lint-locale-dependence.py)
```
Comment (knst) [inline doc/developer-notes.md:850]: 27220: we don't have `lint-locale-dependence.py` yet; can be done after bitcoin/bitcoin#24932 is backported

Comment (knst) [inline doc/developer-notes.md:850]: 27220: we don't have `lint-locale-dependence.py` yet; can be done after bitcoin/bitcoin#24932 is backported

Comment (PastaPastaPasta) [reply doc/developer-notes.md]: dropped

File: contrib/guix/manifest.scm [hunk 6]
```diff
@@ -28,6 +28,7 @@
              (gnu packages bison)
              (gnu packages tls)
              (gnu packages version-control)
+             (guix build-system cmake)
```
Comment (knst) [inline contrib/guix/manifest.scm:30]: 27779, 27179 - need to check GUIX build succeed

**Review by knst — APPROVED** [review]
utACK 7a3c56f12c004c8a0cde5161e6960c32aae30f5e

**Comment by UdjinM6** [comment]
guix fails

File: contrib/guix/manifest.scm (hunk 6 above)
Comment (PastaPastaPasta) [reply contrib/guix/manifest.scm]: dropped both for now. May investigate more which caused the issue

**Review by knst — APPROVED** [review]
utACK 0dbafcee46a45a94d6b99edb9d72906c5e86597e

**Review by UdjinM6 — APPROVED** [review]
utACK 0dbafcee46a45a94d6b99edb9d72906c5e86597e

**Pull request merged**